aspen/logging.py

```diff
@@ -25,51 +10,33 @@
 from __future__ import print_function
 from __future__ import unicode_literals
 
-from __future__ import with_statement
-import os
-import pprint
 import sys
-import thread
-import threading
-
-
-LOGGING_THRESHOLD = -1
-PID = os.getpid()
-LOCK = threading.Lock()
-
-
-def stringify(o):
-    """Given an object, return a str, never raising ever.
-    """
-    if isinstance(o, str):
-        o = o
-    elif isinstance(o, unicode):
-        o = o.encode('UTF-8', 'backslashreplace')
-    else:
-        o = pprint.pformat(o)
-    return o
+import pprint
+import logging
 
+LOGGING_THRESHOLD = 0
 
 def log(*messages, **kw):
-    level = kw.get('level', 0)
-    if level >= LOGGING_THRESHOLD:
-        # Be sure to use Python 2.5-compatible threading API.
-        t = threading.currentThread()
-        fmt = "pid-%s thread-%s (%s) %%s" % ( PID
-                                            , thread.get_ident()
-                                            , t.getName()
-                                             )
-        for message in messages:
-            message = stringify(message)
-            for line in message.splitlines():
-                with LOCK:
-                    # Log lines can get interleaved, but that's okay, because
-                    # we prepend lines with thread identifiers that can be used
-                    # to reassemble log messages per-thread.
-                    print(fmt % line.decode('utf8'))
-                    sys.stdout.flush()
+    """
+    Make logging more convenient - use magic to get the __name__ of the calling module/function
```

Comment:
It comes from an ancient version of `web.py` where it was encoded in the `upvars()` function, but also from http://stackoverflow.com/questions/14597499/function-to-print-current-modules-name-also-when-called-from-another-module/ ; `sys._getframe` is [documented](https://docs.python.org/2.7/library/sys.html#sys._getframe) but is C-python only; I thought that was a reasonable limitation.  If not, we'll have to either: 1) settle on a single logger ('aspen' ?) for the whole library or 2) fix up all calls to `log` and `log_dammit` to use python `logging` directly.